src/__tests__/types.test.ts

```diff
@@ -392,6 +392,172 @@ describe('types', () => {
       });
     });
 
+    describe('tuple', () => {
```

Comment:
nit: move this below to keep these sections alphabetical within the `complex types` section.